python/mmSolver/utils/raytrace.py

```diff
@@ -92,8 +93,20 @@ def closest_intersect(source,
     min_dist = max_dist
     for point in hit_points:
         dist = source_pt.distanceTo(point)
+        dot = (point.x * source_pt.y) - (point.y * source_pt.x)
+        if dot > 0:  # Same
+            sign = 1
+        elif dot < 0:  # Opposite
+            sign = -1
+        else:  # Perpendicular
+            sign = 0
+
+        if sign == 1:
```

Comment:
If the user wants to `test_both_directions` and Maya is will not respect the correct value, we should always pass `test_both_directions=True` to `MFnMesh.closestIntersection`, and then only accept points with the same direction as our input `direction_vec`.

For example, don't add a "continue" in this code block, but instead add it before adding to the `hit_points` list.